applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -378,105 +438,432 @@ static int cmd_list(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
-static int cmd_start(const struct shell *shell, size_t argc, char **argv)
+static int adv_create_and_start(const struct shell *shell, uint8_t big_index)
 {
 	int ret;
 
 	size_t ext_adv_buf_cnt = 0;
 	size_t per_adv_buf_cnt = 0;
 
-	if (argc < 2) {
-		shell_error(shell, "Usage: bct start <preset>");
-		return -EINVAL;
+	/* Get advertising set for BIG0 */
+	ret = ext_adv_populate(big_index, &ext_adv_data[big_index], ext_adv_buf[big_index],
+			       ARRAY_SIZE(ext_adv_buf[big_index]), &ext_adv_buf_cnt);
+	if (ret) {
+		return ret;
 	}
 
-	struct bt_bap_lc3_preset *preset = preset_find(argv[1]);
+	ret = per_adv_populate(big_index, &per_adv_data[big_index], &per_adv_buf[big_index], 1,
+			       &per_adv_buf_cnt);
+	if (ret) {
+		return ret;
+	}
 
-	if (!preset) {
-		shell_error(shell, "Preset not found");
-		return -EINVAL;
+	/* Start broadcaster */
+	ret = bt_mgmt_adv_start(big_index, ext_adv_buf[big_index], ext_adv_buf_cnt,
+				&per_adv_buf[big_index], per_adv_buf_cnt, false);
+	if (ret) {
+		shell_error(shell, "Failed to start advertising for BIG%d", big_index);
+		return ret;
 	}
 
-	struct broadcast_source_big broadcast_param;
+	return 0;
+}
 
-	broadcast_create(&broadcast_param, preset);
+static int cmd_start(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
 
-	ret = broadcast_source_enable(&broadcast_param, 1);
-	if (ret) {
-		shell_error(shell, "Failed to enable broadcaster(s): %d", ret);
-		return ret;
+	if (argc == 2) {
```

Comment:
I think the checks here must be a bit stricter.

If argc > 2 then the else will execute. There should likely be an error.

Also, there is no guarantee that argv[1] is a number. A `is_digit` check should be done.

Reply:
We do check it by doing stroul and then comparing to `CONFIG_BT_ISO_MAX_BIG`. I canadd a check that if argc > 2 then print a usage helper

Reply:
Can't use isdigit() since that takes in one char, not a char pointer